lib/instream/connection/query_runner.ex

```diff
@@ -12,13 +17,33 @@ defmodule Instream.Connection.QueryRunner do
   Executes `:ping` queries.
   """
   @spec ping(Query.t, Keyword.t, Keyword.t) :: :pong | :error
-  def ping(%Query{} = query, _opts, conn) do
+  def ping(%Query{} = query, opts, conn) do
     headers = conn |> Headers.assemble()
 
-    conn
-    |> URL.ping(query.opts[:host])
-    |> :hackney.head(headers, "", Keyword.get(conn, :http_opts, []))
-    |> Response.parse_ping()
+    { query_time, response } = :timer.tc fn ->
+      conn
+      |> URL.ping(query.opts[:host])
+      |> :hackney.head(headers, "", Keyword.get(conn, :http_opts, []))
+    end
+
+    result = response |> Response.parse_ping()
+    status = case response do
+      { :ok, status, _ } -> status
+      _                  -> 0
```

Comment:
Why not `nil` or `:error` or even the error itself `{:error, reason}`? I didn't check what response can be though.